pkg/scaler/prober.go

```diff
@@ -527,3 +537,33 @@ func (p *prober) scaleUp(ctx context.Context) error {
 		return n > o // scale to at least n
 	})
 }
+
+// verifies if for the for the respecitve scale operations the depenendent deployments are in desired state before scaling the actual deployments
+func (p *prober) checkScaleRefDependsOn(ctx context.Context, prefix string, dependsOnScaleRefs []autoscalingapi.CrossVersionObjectReference, replicas int32, checkFn func(oReplicas, nReplicas int32) bool) bool {
+	// if possible check from the cache if the target needs to be scaled
+	klog.V(4).Infof("Check scale for dependents with prefix %s and dependendents %v", prefix, dependsOnScaleRefs)
+	if len(dependsOnScaleRefs) != 0 {
+		for _, dependsOnScaleRef := range dependsOnScaleRefs {
+			klog.V(4).Infof("Checking if the depenendent scaleRef %v  has the desired replicas %d\n ", dependsOnScaleRef, replicas)
+			if dependsOnScaleRef.APIVersion == appsv1.SchemeGroupVersion.String() && dependsOnScaleRef.Kind == kindDeployment {
+				dwdGetTargetFromCacheTotal.With(prometheus.Labels{labelResource: resourceDeployments}).Inc()
+				d, err := p.deploymentsLister.Deployments(p.namespace).Get(dependsOnScaleRef.Name)
+				if err != nil {
+					klog.Errorf("%s: Could not find the target reference for %s: %s", prefix, dependsOnScaleRef.Name, err)
+					return false
+				}
+				var availableReplicas = int32(0)
+				availableReplicas = d.Status.AvailableReplicas //check if available replicas is as desired
+				if !checkFn(availableReplicas, replicas) {
+					klog.V(4).Infof("%s: skipped because desired=%d and available=%d", prefix, replicas, availableReplicas)
```

Comment:
This log will print `skipped` even when desired == available. So it needs to be improved. for example check the log you pasted in conversation at time `06:59:20.054548`